### DISCUSSION TITLE
Legal problem? If using valkey, Redis company can sue ?

### DISCUSSION THREAD

**Discussion by code959437957** [valkey-io/valkey#69] [General]
Legal problem? If using valkey, Redis company can sue ?
It is legal to safe using valkey by law?

Not be sue by Redis company?

**Comment by natoscott** [comment]
@code959437957 Can you expand on your concern?  On what basis would they sue you for using valkey, which is a fork from the last version of redis under BSD license?  (the license change cannot effect earlier versions of the source code - if legal, it would apply only from the time the license change was made)

**Comment by dmitrypol** [comment]
Since Valkey is forked from Redis 7.2.4 which still has old BSD license I can't see how they can sue (I am not a lawyer).

**Reply by SoldierTaker** [reply]
I agree. The licensing would have to be up to date for that specific version for them to be able to sue. As long as wasn’t during the time when that BSD license was the most recent one, they really can’t do anything about it, but make empty threats.

**Comment by zuiderkwast** [comment]
Regarding the license, we're safe.

Regarding the trademark, we'll make sure to be legal. We shall not to use the name Redis. When it comes to API compatibility, it is considered fair use when the word "Redis" is part of an API or protocol.

**Reply by kirmorozov** [reply]
When it comes to API compatibility, follow Oracle vs Google precedent. 
Moreover refer that wire protocol is not a subject for license.
Clickhouse and AWS Aurora both are MySQL and Postgess Wire compatible.

**Comment by hairmare** [comment]
in summary

* yes, it is possible that Redis Inc. decides to take legal action against projects like ValKey
* there is a distinction to be made between the Redis protocol and it's implementation, for most users this does not matter
* should Redis try to sue the ValKey project, the Linux Foundation will help with legal efforts

I think in regards to the question raised, the ValKey project has got it's bases covered.

please close this issue if this addresses your concerns or let the community know what additional assurances your use-case requires.
